lib/rucio/client/baseclient.py

```diff
@@ -157,11 +157,14 @@ def __init__(self,
         rucio_scheme = urlparse(self.host).scheme
         auth_scheme = urlparse(self.auth_host).scheme
 
-        if rucio_scheme != 'http' and rucio_scheme != 'https':
-            raise ClientProtocolNotSupported('\'%s\' not supported' % rucio_scheme)
+        rucio_scheme_allowed = ['http', 'https']
+        auth_scheme_allowed = ['http', 'https']
 
-        if auth_scheme != 'http' and auth_scheme != 'https':
-            raise ClientProtocolNotSupported('\'%s\' not supported' % auth_scheme)
+        if rucio_scheme not in rucio_scheme_allowed:
+            raise ClientProtocolNotSupported(f"Scheme '{rucio_scheme}' not supported when connecting to Rucio Server '{self.host}'. Allowed schemes: {rucio_scheme_allowed}")
+
+        if auth_scheme not in auth_scheme_allowed:
+            raise ClientProtocolNotSupported(f"Scheme '{auth_scheme}' not supported when connecting to Rucio Auth '{self.host}'. Allowed schemes: {auth_scheme_allowed}")
```

Comment:
For clarity, should we also have a separate exception message raised when either scheme is not passed?

Reply:
I agree, I can add it in this PR.

Another approach would be to default to `https` in case the protocol is missing. I'm not sure if there are any side effects this could have though.